syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "custom_types.proto";

service NotificationService {
  rpc GetAll (GetAllRequest) returns (GetAllNotificationsResponse);
  rpc GetById (GetNotificationByIdRequest) returns (NotificationModel);
  rpc GetByDateRange (GetByDateRangeRequest) returns (GetAllNotificationsResponse);
  rpc GetByCustomer (GetNotificationsByCustomerRequest) returns (GetAllNotificationsResponse);
}

message NotificationModel {
  int32 notification_id = 1;
  optional int32 customer_id = 2;
  optional string message = 3;
  optional google.protobuf.Timestamp created_at = 4;
  optional bool is_read = 5;
}

message GetAllNotificationsResponse {
  repeated NotificationModel notifications = 1;
}

message GetNotificationByIdRequest {
  int32 notification_id = 1;
}

message GetNotificationsByCustomerRequest {
  int32 customer_id = 1;
  optional bool is_read = 2;
}
